tests/test_pagure.py

```diff
@@ -0,0 +1,749 @@
+
+    def test_has_archiving(self):
+        """Test if it returns True when has_archiving is called"""
+
+        self.assertEqual(Pagure.has_archiving(), True)
+
+    @httpretty.activate
+    def test_fetch_issues(self):
+        """Test whether a list of issues is returned"""
+
+        body = read_file('data/pagure/pagure_repo_issue_1')
+
+        httpretty.register_uri(httpretty.GET,
+                               PAGURE_ISSUES_URL,
+                               body=body,
+                               status=200,
+                               )
+        pagure = Pagure(repository='Project-example', api_token='aaa')
+        issues = [issues for issues in pagure.fetch(from_date=None, to_date=None)]
+
+        self.assertEqual(len(issues), 1)
+
+        issue = issues[0]
+        self.assertEqual(issue['origin'], 'https://pagure.io/Project-example')
+        self.assertEqual(issue['uuid'], '41071b08dd75f34ca92c6d5ecb844e7a3e5939c6')
+        self.assertEqual(issue['updated_on'], 1583508642.0)
+        self.assertEqual(issue['category'], CATEGORY_ISSUE)
+        self.assertEqual(issue['tag'], 'https://pagure.io/Project-example')
+        self.assertEqual(len(issue['data']['comments']), 1)
+        self.assertEqual(issue['data']['comments'][0]['user']['name'], 'animeshk08')
+
+    @httpretty.activate
+    def test_fetch_issues_disabled(self):
+        """Test whether an empty response is returned in case a repository
+           has disabled issue tracker
+        """
+        body = read_file('data/pagure/pagure_empty_request')
+
+        httpretty.register_uri(httpretty.GET,
+                               PAGURE_ISSUES_URL,
+                               body=body,
+                               status=404,
+                               )
+        pagure = Pagure(repository='Project-test-example')
+        issues = [issues for issues in pagure.fetch(from_date=None, to_date=None)]
+        self.assertEqual(len(issues), 0)
+        self.assertListEqual(issues, [])
+
+    @httpretty.activate
+    def test_search_fields_issues(self):
+        """Test whether the search_fields is properly set"""
+
+        body = read_file('data/pagure/pagure_repo_issue_1')
+
+        httpretty.register_uri(httpretty.GET,
+                               PAGURE_ISSUES_URL,
+                               body=body,
+                               status=200,
+                               )
+
+        pagure = Pagure(repository='Project-example', api_token='aaa')
+        issues = [issues for issues in pagure.fetch(from_date=None, to_date=None)]
+
+        issue = issues[0]
+        self.assertEqual(pagure.metadata_id(issue['data']), issue['search_fields']['item_id'])
+        self.assertIsNone(issue['search_fields']['namespace'])
+        self.assertEqual(issue['search_fields']['repo'], 'Project-example')
+
+    @httpretty.activate
+    def test_fetch_issues_reaction(self):
+        """Test the reactions on an issue comment"""
```

Comment:
I'm not sure this test is needed, since the reactions are directly included in the comments (we don't need any extra call to the API).